pkg/docker/docker_api.go

```diff
@@ -0,0 +1,283 @@
+	values = append(values, []interface{}{dockLogs.LastUpdated, dockLogs.PullCount, dockLogs.StarCount})
+
+	rangeValue := dockLogs.Name + "!A:C"
+	valueInputOption := "RAW"
+	rb := &sheets.ValueRange{
+		Values: values,
+	}
+
+	resp, err := srv.Spreadsheets.Values.Append(SpreadSheetId, rangeValue, rb).ValueInputOption(valueInputOption).Context(ctx).Do()
+
+	if err != nil {
+		return "", errors.FromErr(err).Err()
+	}
+	return fmt.Sprintf("Successful [%v] row insertion in [%v] for [%v/%v]", resp.Updates.UpdatedRows, resp.SpreadsheetId, dockLogs.User, dockLogs.Name), nil
+}
+
+func processAndUpdate(spreadSheetId string, link string) {
+	dockerResp, err := getDockerLogs(link)
+	if err != nil {
+		log.Fatalf("Unable to retrieve Docker log.s %v", err)
+	}
+
+	for _, c := range dockerResp.Results {
+		respStr, err := updateSheet(DockerRepoLogs{
+			Name:        c.Name,
+			User:        c.User,
+			StarCount:   c.StarCount,
+			PullCount:   c.PullCount,
+			LastUpdated: c.LastUpdated,
+		}, spreadSheetId)
+
+		if err != nil {
+			log.Println("Error while updating ", c.Name, err)
+		} else {
+			log.Println(respStr)
+		}
+	}
+
+	log.Println(dockerResp.Next)
+	for str, ok := dockerResp.Next.(string); ok; {
+		dockerResp, err = getDockerLogs(str)
+
+		if err != nil {
+			log.Fatalf("Unable to retrieve Docker log.s %v", err)
+		}
+		for _, c := range dockerResp.Results {
+			respStr, err := updateSheet(DockerRepoLogs{
+				Name:        c.Name,
+				User:        c.User,
+				StarCount:   c.StarCount,
+				PullCount:   c.PullCount,
+				LastUpdated: c.LastUpdated,
+			}, spreadSheetId)
+			if err != nil {
+				log.Println("Error while updating ", c.Name, err)
+			} else {
+				log.Println(respStr)
+			}
+		}
+		str, ok = dockerResp.Next.(string)
+		log.Println(dockerResp.Next)
+	}
+}
+
+// File Path: ~/.credentials/client_secret_spreadsheet.json
+func getSecretFilePath() (string, error) {
+	usr, err := user.Current()
+	if err != nil {
+		return "", err
+	}
+	fileDir := filepath.Join(usr.HomeDir, ".credentials")
+	os.MkdirAll(fileDir, 0700)
+	return filepath.Join(fileDir,
+		url.QueryEscape("client_secret_spreadsheet.json")), err
+}
+
+func deleteSheet(SpreadSheetId string) {
+	ctx := context.Background()
+
+	//Reading Secret File from /.Credential
+	secretFilePath, err := getSecretFilePath()
+	if err != nil {
+		log.Println("Unable to detect repository of client secret file")
+	}
+	b, err := ioutil.ReadFile(secretFilePath)
+	if err != nil {
+		log.Println("Unable to read client secret file")
+	}
+
+	// If modifying these scopes, delete your previously saved credentials
+	// at ~/.credentials/sheets.googleapis.com-go-quickstart.json
+	config, err := google.ConfigFromJSON(b, "https://www.googleapis.com/auth/spreadsheets")
+	if err != nil {
+		log.Println("Unable to parse client secret file to config")
+	}
+	client := spreadsheet.GetClient(ctx, config)
+	srv, err := sheets.New(client)
+	if err != nil {
+		log.Println("Unable to read client secret file: ")
+	}
+
```

Comment:
remove the previous repetation. move them to a separate method or function and use them as where necessary.